zirgen/Dialect/BigInt/IR/Eval.cpp

```diff
@@ -183,15 +183,20 @@ Digest computeDigest(std::vector<BytePoly> witness, size_t groupCount) {
   return poseidon2Hash(words.data(), words.size());
 }
 
-EvalOutput eval(func::FuncOp inFunc, ArrayRef<APInt> witnessValues) {
+struct Def {
+  virtual APInt load(uint32_t arena, uint32_t offset, uint32_t count) = 0;
+  virtual void store(uint32_t arena, uint32_t offset, uint32_t count, APInt val) = 0;
+};
+
+EvalOutput eval(func::FuncOp inFunc, BigIntIO& io, bool computeZ) {
   EvalOutput ret;
 
   llvm::DenseMap<Value, BytePoly> polys;
 
   for (Operation& origOp : inFunc.getBody().front().without_terminator()) {
     llvm::TypeSwitch<Operation*>(&origOp)
         .Case<DefOp>([&](auto op) {
-          APInt val = witnessValues[op.getLabel()];
```

Comment:
if DefOp is just another way of calling `.load`, perhaps we should drop it from the dialect?